hub/config/src/test/java/uk/gov/ida/hub/config/domain/CertificateChainConfigValidatorTest.java

```diff
@@ -0,0 +1,92 @@
+
+@RunWith(MockitoJUnitRunner.class)
+public class CertificateChainConfigValidatorTest {
+
+    @Mock
+    private TrustStoreForCertificateProvider trustStoreForCertificateProvider;
+
+    @Mock
+    private CertificateChainValidator certificateChainValidator;
+
+    @Mock
+    private KeyStore trustStore;
+
+    private CertificateChainConfigValidator validator;
+
+    @Before
+    public void setUp() {
+        when(trustStoreForCertificateProvider.getTrustStoreFor(FederationEntityType.RP)).thenReturn(trustStore);
+
+        validator = new CertificateChainConfigValidator(trustStoreForCertificateProvider, certificateChainValidator);
+    }
+
+    @Test
+    public void validateLogsErrorForMissingCertificatesWhenNotSelfService() {
+        Logger logger = (Logger) LoggerFactory.getLogger(CertificateChainConfigValidator.class);
+        ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
+
+        TransactionConfig config = aTransactionConfigData().withEncryptionCertificate("").withSelfService(false).build();
+        Certificate signingCert = config.getSignatureVerificationCertificates().stream().findFirst().get();
+
+        when(certificateChainValidator.validate(signingCert.getX509Certificate().get(), trustStore)).thenReturn(CertificateValidity.valid());
+
+        validator.validate(Set.of(config));
+
+        List<ILoggingEvent> logsList = listAppender.list;
+        assertThat(logsList.get(0).getLevel().toString()).isEqualTo(Level.ERROR.toString());
+
+    }
+
+    @Test
+    public void validateLogsWarningForX509CertsThatFailChainValidation() {
+        Logger logger = (Logger) LoggerFactory.getLogger(CertificateChainConfigValidator.class);
+        ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+        listAppender.start();
+        logger.addAppender(listAppender);
+
+        TransactionConfig config = aTransactionConfigData().withSelfService(false).build();
+        Certificate encryptionCert = config.getEncryptionCertificate();
+        Certificate signingCert = config.getSignatureVerificationCertificates().stream().findFirst().get();
+
+        CertPathValidatorException certPathValidatorException = new CertPathValidatorException("Bad Cert Chain");
+
+        when(certificateChainValidator.validate(encryptionCert.getX509Certificate().get(), trustStore)).thenReturn(CertificateValidity.invalid(certPathValidatorException));
+        when(certificateChainValidator.validate(signingCert.getX509Certificate().get(), trustStore)).thenReturn(CertificateValidity.valid());
+
+        validator.validate(Set.of(config));
+
+        List<ILoggingEvent> logsList = listAppender.list;
+        assertThat(logsList.get(0).getLevel().toString()).isEqualTo(Level.WARN.toString());
+        assertThat(logsList.get(0).getMessage()).contains("Bad Cert Chain");
+
+    }
+
```

Comment:
Might be worth testing that we *don't* log anything when a cert is present and valid (both for self-service and for local).

Reply:
Yup, good idea.